cmd/monaco/download/download_command.go

```diff
@@ -30,101 +30,92 @@ import (
+
+	// download via manifest
+	cmd.Flags().StringVarP(&f.manifestFile, "manifest", "m", "manifest.yaml", "Name (and the path) to the manifest file. If not provided \"manifest.yaml\" value will be used.")
+	cmd.Flags().StringVarP(&f.specificEnvironmentName, "environment", "e", "", "Specify a concrete environment defined in the manifest file that shall be downloaded")
+	// download without manifest
+	cmd.Flags().StringVar(&f.environmentURL, "url", "", "URL to the dynatrace environment from which to download configuration from. To be able to connect, token and, in case of connecting to platform, a pari of OAuth client ID na client secret needs to bi provide via adequate flags (\"--token\", \"--oauth-client-id\", \"--oauth-client-secret\"). Not able to combine with \"--manifest\".")
+	cmd.Flags().StringVar(&f.token, "token", "", "Token secret to connect to DT server. Use only with \"--url\"")
+	cmd.Flags().StringVar(&f.clientID, "oauth-client-id", "", "OAuth client ID is used to connect to DT server via OAuth. Use only with \"--url\"")
+	cmd.Flags().StringVar(&f.clientSecret, "oauth-client-secret", "", "OAuth client secret is used to connect to DT server via OAuth. Use only with \"--url\"")
+
+	// download options
+	cmd.Flags().StringSliceVarP(&f.specificAPIs, "api", "a", nil, "One or more APIs to download (flag can be repeated or value defined as comma-separated list)")
+	cmd.Flags().StringSliceVarP(&f.specificSchemas, "settings-schema", "s", nil, "One or more settings 2.0 schemas to download (flag can be repeated or value defined as comma-separated list)")
+
+	cmd.Flags().BoolVar(&f.onlyAPIs, "only-apis", false, "Only download config APIs, skip downloading settings 2.0 objects")
+	cmd.Flags().BoolVar(&f.onlySettings, "only-settings", false, "Only download settings 2.0 objects, skip downloading config APIs")
+	cmd.MarkFlagsMutuallyExclusive("settings-schema", "only-apis", "only-settings")
+	cmd.MarkFlagsMutuallyExclusive("api", "only-apis", "only-settings")
+	cmd.MarkFlagsMutuallyExclusive("only-apis", "only-settings")
+
 	if featureflags.Entities().Enabled() {
-		getDownloadEntitiesCommand(fs, command, downloadCmd)
+		getDownloadEntitiesCommand(fs, command, cmd)
+	}
+
+	err := cmd.RegisterFlagCompletionFunc("api", completion.AllAvailableApis)
+	if err != nil {
+		log.Fatal("failed to setup CLI %v", err)
+	}
+
+	return cmd
+}
+
+func preRunChecks(f downloadCmdOptions) error {
+	switch {
+	case f.environmentURL != "" && f.manifestFile != "manifest.yaml":
+		return errors.New("\"url\" and \"token\" are mutually exclusive")
+	case f.environmentURL != "" && f.specificEnvironmentName != "":
+		return errors.New("")
+	case f.environmentURL != "":
+		switch {
+		case f.token == "":
+			return errors.New("if \"url\" is set, \"token\" also must be set")
+		case (f.clientID == "") != (f.clientSecret == ""):
+			return errors.New("\"oauth-client-id\" and \"oauth-client-secret\" always come in pair")
```

Comment:
```suggestion
			return errors.New("\"oauth-client-id\" and \"oauth-client-secret\" must always be set together")
```

Reply:
actually I wonder if the base cobra `RequiredTogether` wouldn't give us a better message for this - IIRC it tells you which one you're missing

Reply:
It would be nice if there is an way to say order of the rules. E.g. for setting that `manifest` and `token` are mutually exclusive, and that `token` and `url` needs to come together, in case of the call:
```
monaco download --manifest myManifest.yaml --token TOKEN...
``` 
error message will be like: `token` and `url` must be both set. 

For `... --manifest .. --url --token` error will report that `manifest` and `url` are forbidden combination.


That is by my opinion misleading